src/coreclr/pal/src/exception/machexception.cpp

```diff
@@ -369,28 +369,22 @@ void PAL_DispatchException(PCONTEXT pContext, PEXCEPTION_RECORD pExRecord, MachE
 {
     CPalThread *pThread = InternalGetCurrentThread();
 
-    CONTEXT *contextRecord;
-    EXCEPTION_RECORD *exceptionRecord;
-    AllocateExceptionRecords(&exceptionRecord, &contextRecord);
+    CONTEXT *contextRecord = pContext;
+    g_hardware_exception_context_locvar_offset = (int)((char*)&contextRecord - (char*)__builtin_frame_address(0));
 
-    *contextRecord = *pContext;
-    *exceptionRecord = *pExRecord;
-
-    contextRecord->ContextFlags |= CONTEXT_EXCEPTION_ACTIVE;
+    pContext->ContextFlags |= CONTEXT_EXCEPTION_ACTIVE;
     bool continueExecution;
-
     {
-        // The exception object takes ownership of the exceptionRecord and contextRecord
-        PAL_SEHException exception(exceptionRecord, contextRecord);
+        PAL_SEHException exception(pExRecord, pContext, true);
```

Comment:
I see the `onStack` argument is being set to `true` here. This appears to be a fundamental assumption change in this code path. Is a comment warranted in this case?

Reply:
The `onStack` means that we are constructing the exception from the on stack context and so when the `PAL_SEHException` is destroyed, we don't attempt to free that memory. That's why I have removed the comment above saying that it takes the ownership. I'm not sure what comment you'd like to see added here. We were constructing it from an on-heap copy of the records that were allocated right before the call before my change.